flow/flow.go

```diff
@@ -0,0 +1,303 @@
+package flow
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/micro/go-micro/v2/broker"
+	"github.com/micro/go-micro/v2/client"
+	"github.com/micro/go-micro/v2/logger"
+)
+
+type Flow interface {
+	// Init flow with options
+	Init(...Option) error
+	// Get flow options
+	Options() Options
+	// Create step in specific flow
+	CreateStep(flow string, step *Step) error
+	// Delete step from specific flow
+	DeleteStep(flow string, step *Step) error
+	// Replace step in specific flow
+	ReplaceStep(flow string, oldstep *Step, newstep *Step) error
+	// Lookup specific flow
+	Lookup(flow string) ([]*Step, error)
+	// Execute specific flow and returns request id and error, optionally fills rsp
+	Execute(flow string, req interface{}, rsp interface{}, opts ...ExecuteOption) (string, error)
+	// Resume specific paused flow execution by request id
+	Resume(flow string, reqID string) error
+	// Pause specific flow execution by request id
+	Pause(flow string, reqID string) error
+	// Abort specific flow execution by request id
+	Abort(flow string, reqID string) error
+	// State show state specific flow execution by request id
+	State(flow string, reqID string) (string, error)
+	// Result get result of the flow step
+	Result(flow string, reqID string, step *Step) ([]byte, error)
+	// Stop executor and drain active workers
+	Stop() error
+}
+
+type Step struct {
+	// name of step
+	ID string
+	// Retry count for step
+	Retry int
```

Comment:
Retry/Timeout feel like options?

Reply:
For some steps you want to have multiple retries, for some - only one.

Reply:
They should be options to the Create/Add method

Reply:
in this case when i'm execute step i need to know this variables and if i don't put it into struct, i need to use context. I don't this that adding context to Step struct is really needed